src/app/modules/form/admin-form/admin-form.service.ts

```diff
@@ -1356,14 +1356,21 @@ export const deleteTwilioCredentials = (
         SecretId: msgSrvcName,
       })
       .promise(),
-    () => {
+    (error) => {
       logger.info({
         message: 'Twilio Credentials do not exist in Secrets Manager',
         meta: logMeta,
       })
 
-      return new SecretsManagerNotFoundError(
-        'Twilio Credentials do not exist in Secrets Manager',
+      const awsError = error as AWSError
+
+      if (awsError.code === 'ResourceNotFoundException')
+        return new SecretsManagerNotFoundError(
+          'Twilio Credentials do not exist in Secrets Manager',
+        )
+
+      return new SecretsManagerError(
+        'Error occurred when updating Twilio in Secret Manager!',
```

Comment:
should we move the logger.info part into the resourcenotfound block and actually log an error with logger.error in the generic error location?

Reply:
Sure